src/celeritas/ext/VecgeomTrackView.hh

```diff
@@ -363,17 +373,16 @@ CELER_FUNCTION Propagation VecgeomTrackView::find_next_step(real_type max_step)
 
 //---------------------------------------------------------------------------//
 /*!
- * Find the safety at the current position.
+ * Find the nearest distance to *any* boundary.
+ *
+ * \note We might be able to optimize whether to update the safety distance,
+ * since some of the use cases
  */
 CELER_FUNCTION real_type VecgeomTrackView::find_safety()
 {
     CELER_EXPECT(!this->is_outside());
-    real_type safety = detail::BVHNavigator::ComputeSafety(
-        detail::to_vector(this->pos()), vgstate_);
 
-    // TODO: ComputeSafety can return negative safety distances: clamp it to
-    // zero until we debug the underlying cause.
```

Comment:
@sethrj  Just FYI: VecGeom returns negative safeties by design, so let me explain.

VecGeom has two safety algorithms, SafetyToIn() and SafetyToOut(), which are supposed to be called for points outside and inside, respectively.  In some situations, those functions are called from the wrong side, mostly because of the surface tolerance of O(1.e-8cm), and that's when the function returns a negative value.

This is actually makes sense if you think of safety decreasing as a track approaching a surface, until it reaches zero when track is *on surface*, and becomes negative as the track keeps going. So from the VecGeom point of view, handling the negative safety is up to the client code.